### PR TITLE
[mongo-c-driver] bump versions, remove openssl fix

### PR DESCRIPTION
**Pull request by mathbunnyru**
Specify library name and version:  **lib/1.0**

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/mongo-c-driver/all/conandata.yml [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
+    url: "https://github.com/mongodb/mongo-c-driver/releases/download/1.17.4/mongo-c-driver-1.17.4.tar.gz"
+    sha256: "9ec8fe7fb54d636886fa823460658ccf660e3d82520d10810fb7c9d302ac974f"
   "1.17.3":
     url: "https://github.com/mongodb/mongo-c-driver/releases/download/1.17.3/mongo-c-driver-1.17.3.tar.gz"
     sha256: "6594cbae17389005bcac5a8d4091af3be4894d5314a519504d4b4509effcc0df"
   "1.17.2":
     url: "https://github.com/mongodb/mongo-c-driver/releases/download/1.17.2/mongo-c-driver-1.17.2.tar.gz"
     sha256: "bc53d5f72ab628a1ae549db6888325d6dc34db3ca31c5e16e550c1bb4266d864"
+    url: "https://github.com/mongodb/mongo-c-driver/releases/download/1.17.2/mongo-c-driver-1.17.2.tar.gz"
+    sha256: "bc53d5f72ab628a1ae549db6888325d6dc34db3ca31c5e16e550c1bb4266d864"
```
Comment (SpaceIm) [inline recipes/mongo-c-driver/all/conandata.yml:12]:
```suggestion
```

**Comment by SpaceIm** [comment]
```
sicuucd.lib(umutex.ao) : error LNK2019: unresolved external symbol __imp___std_init_once_begin_initialize referenced in function "void __cdecl std::call_once<void (__cdecl&)(void)>(struct std::once_flag &,void (__cdecl&)(void))" (??$call_once@A6AXXZ$$V@std@@YAXAEAUonce_flag@0@A6AXXZ@Z) [C:\J\w\cci_PR-4752\s\b6e685\1\build_subfolder\source_subfolder\src\libmongoc\mongoc_shared.vcxproj]
sicuucd.lib(umutex.ao) : error LNK2019: unresolved external symbol __imp___std_init_once_complete referenced in function "public: __cdecl std::_Init_once_completer::~_Init_once_completer(void)" (??1_Init_once_completer@std@@QEAA@XZ) [C:\J\w\cci_PR-4752\s\b6e685\1\build_subfolder\source_subfolder\src\libmongoc\mongoc_shared.vcxproj]
C:\J\w\cci_PR-4752\s\b6e685\1\build_subfolder\bin\mongoc-1.0.dll : fatal error LNK1120: 2 unresolved externals [C:\J\w\cci_PR-4752\s\b6e685\1\build_subfolder\source_subfolder\src\libmongoc\mongoc_shared.vcxproj]
mongo-c-driver/1.17.4: 
```

It seems to be an issue in icu recipe.

https://docs.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-initoncebegininitialize#remarks

**Comment by mathbunnyru** [comment]
> ```
> sicuucd.lib(umutex.ao) : error LNK2019: unresolved external symbol __imp___std_init_once_begin_initialize referenced in function "void __cdecl std::call_once<void (__cdecl&)(void)>(struct std::once_flag &,void (__cdecl&)(void))" (??$call_once@A6AXXZ$$V@std@@YAXAEAUonce_flag@0@A6AXXZ@Z) [C:\J\w\cci_PR-4752\s\b6e685\1\build_subfolder\source_subfolder\src\libmongoc\mongoc_shared.vcxproj]
> sicuucd.lib(umutex.ao) : error LNK2019: unresolved external symbol __imp___std_init_once_complete referenced in function "public: __cdecl std::_Init_once_completer::~_Init_once_completer(void)" (??1_Init_once_completer@std@@QEAA@XZ) [C:\J\w\cci_PR-4752\s\b6e685\1\build_subfolder\source_subfolder\src\libmongoc\mongoc_shared.vcxproj]
> C:\J\w\cci_PR-4752\s\b6e685\1\build_subfolder\bin\mongoc-1.0.dll : fatal error LNK1120: 2 unresolved externals [C:\J\w\cci_PR-4752\s\b6e685\1\build_subfolder\source_subfolder\src\libmongoc\mongoc_shared.vcxproj]
> ```
> 
> 
> https://docs.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-initoncebegininitialize#remarks
> 
> 
> Maybe shared MTd is not supported

Thanks, I'll take a look.

**Comment by mathbunnyru** [comment]
The same profile worked on my machine, so I'll try to relaunch.

File: recipes/mongo-c-driver/all/CMakeLists.txt [hunk 2]
```diff
@@ -5,3 +5,9 @@ include(conanbuildinfo.cmake)
 conan_basic_setup()
 
 add_subdirectory(source_subfolder)
+
+if(MSVC)
```
Comment (mathbunnyru) [inline recipes/mongo-c-driver/all/CMakeLists.txt:9]: I will try to remove this patch, it might not be needed.

Comment (mathbunnyru) [reply recipes/mongo-c-driver/all/CMakeLists.txt:9]: This fix was indeed needed

**Comment by SpaceIm** [comment]
I think there is an issue in Visual Studio 16 agent https://github.com/conan-io/conan-center-index/pull/4810#issuecomment-792625533

**Review by uilianries — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**